### PR TITLE
waylogout: init at unstable-2023-06-09

### LINKED ISSUE #242499: waylogout
waylogout is graphical logout/suspend/reboot/shutdown dialog for wayland. It is inspired by [oblogout](https://launchpad.net/oblogout) and based on code from [swaylock-effects](https://github.com/mortie/swaylock-effects).

**Metadata**

* [homepage URL](https://github.com/loserMcloser/waylogout)
* [source URL](https://github.com/loserMcloser/waylogout)
* license: mit
* platforms: linux

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

Closes #242499.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by majiru** [comment]
Result of `nixpkgs-review pr 242774` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>waylogout</li>
  </ul>
</details>

**Review by majiru — APPROVED** [review]

File: pkgs/tools/wayland/waylogout/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+, pkg-config
+, wayland
+, wayland-protocols
+, libxkbcommon
+, cairo
+, gdk-pixbuf
+, scdoc
+}:
+
+stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/tools/wayland/waylogout/default.nix:15]: I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/tools/wayland/waylogout/default.nix (hunk 1 above)
Comment (majiru) [reply pkgs/tools/wayland/waylogout/default.nix:15]: It appears to me that the 'rec' attribute isn't being used here for anything, and instead could be removed altogether.

Comment (dit7ya) [reply pkgs/tools/wayland/waylogout/default.nix]: I removed the `rec`.

**Review by drupol — APPROVED** [review]

**Pull request merged**